text/src/autogluon/text/automm/optimization/lit_distiller.py

```diff
@@ -0,0 +1,384 @@
+        batch_idx
+            Index of mini-batch.
+
+        Returns
+        -------
+        Average loss of the mini-batch data.
+        """
+        _, loss = self._shared_step(batch)
+        self.log("train_loss", loss)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        """
+        Per validation step. This function is registered by pl.LightningModule.
+        Refer to https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#validation
+
+        Parameters
+        ----------
+        batch
+            A dictionary containing the mini-batch data, including both input data and
+            ground-truth labels. The mini-batch data are passed to each individual model,
+            which indexes its required input data by keys with its model prefix. The
+            ground-truth labels are used here to compute the validation loss and metric.
+            The validation metric is used for top k model selection and early stopping.
+        batch_idx
+            Index of mini-batch.
+        """
+        student_output, loss = self._shared_step(batch)
+        # By default, on_step=False and on_epoch=True
+        self.log("val_loss", loss)
+        self.log(
+            self.validation_metric_name,
+            self._compute_metric(
+                logits=student_output[self.student_model.prefix][LOGITS],
+                label=batch[self.student_model.label_key],
+            ),
+        )
+
+    def configure_optimizers(self):
+        """
+        Configure optimizer. This function is registered by pl.LightningModule.
+        Refer to https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers
+        Returns
+        -------
+        [optimizer]
+            Optimizer.
+        [sched]
+            Learning rate scheduler.
+        """
+        if self.hparams.lr_choice == "two_stages":
+            logger.debug("applying 2-stage learning rate...")
+            grouped_parameters = apply_two_stages_lr(
+                model=self.student_model,
+                lr=self.hparams.lr,
+                lr_mult=self.hparams.lr_mult,
+                weight_decay=self.hparams.weight_decay,
+                return_params=True,
+            )
+        elif self.hparams.lr_choice == "layerwise_decay":
+            logger.debug("applying layerwise learning rate decay...")
+            grouped_parameters = apply_layerwise_lr_decay(
+                model=self.student_model,
+                lr=self.hparams.lr,
+                lr_decay=self.hparams.lr_decay,
+                weight_decay=self.hparams.weight_decay,
+            )
+        else:
+            logger.debug("applying single learning rate...")
+            grouped_parameters = apply_single_lr(
+                model=self.student_model,
+                lr=self.hparams.lr,
+                weight_decay=self.hparams.weight_decay,
+            )
```

Comment:
Same parameters passed into all three branches - consider moving those to dict - improves readability and reducing errors possibility:
```python
args = dict(
    model=self.student_model,
    lr=self.hparams.lr,
    weight_decay=self.hparams.weight_decay,
)
if self.hparams.lr_choice == "two_stages":
    logger.debug("applying 2-stage learning rate...")
    grouped_parameters = apply_two_stages_lr(
        lr_mult=self.hparams.lr_mult,
        return_params=True,
        **args,
    )
elif self.hparams.lr_choice == "layerwise_decay":
    logger.debug("applying layerwise learning rate decay...")
    grouped_parameters = apply_layerwise_lr_decay(
        lr_decay=self.hparams.lr_decay,
        **args,
    )
else:
    logger.debug("applying single learning rate...")
    grouped_parameters = apply_single_lr(
        **args,
    )
```

Reply:
Thanks! Done.